revocations.c

```diff
@@ -475,7 +474,7 @@ hibagrl_check(const struct hibagrl *grl, u_int64_t serial, u_int32_t grant_idx)
 
 		if ((min_serial > serial) ||
 		    (max_serial < serial)) {
-			debug2("hibagrl_check: outside of GRL range [0x%" PRIx64 " - 0x%" PRIx64 "]: 0x%" PRIx64,
+			debug2("hibagrl_check: outside of remaining GRL range [0x%" PRIx64 " - 0x%" PRIx64 "]: 0x%" PRIx64,
```

Comment:
This can only be hit in the first iteration of the loop, correct? We should move the check outside.

Reply:
This is actually the loop's exit condition: when the evaluation window does not contain the target serial anymore.
Te min_serial and max_serial are update with the new min and max index at the beginning of each loop.

Reply:
I think what bothers me is that this block is a binary search in a sorted array, but with the exit condition somewhat obfuscated. Couldn't we simply check for `max < min` to the same effect?

Reply:
Done